### PR TITLE
[CU-2xjx1a7] Move benchmarks check to nix

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2xjx1a7

## Description
Also, I removed deprecated CI jobs and Dockerfiles. I temporarily removed `remark` and `hadolint` checks, because they are disabled (`continue-on-error: true`), I'm going to move them to nix in the future.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2xjx1a7 Move benchmarks check to nix](https://app.clickup.com/t/2xjx1a7)

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -932,6 +963,7 @@ jobs:
         - taplo-cli-check
         - prettier-check        
         - check-simnode
+        - benchmarks-check
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:966]: may it was existing `check-benchmarks`?

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -39,35 +39,6 @@ jobs:
...
-          docker system prune --force --all --volumes
-      - uses: actions/checkout@v2
-      - name: Rustup show
-        run: |
-          rustup show
-      - uses: actions/checkout@v2
-      - name: cargo check --all --benches
-        run: |
-          cd code
-          SKIP_WASM_BUILD=1 cargo +nightly check --benches --all --features runtime-benchmarks # checks all rust crates with default-features, tests, benchmarks
```
Comment (dzmitry-lahoda) [inline .github/workflows/check.yml:69]: was used all benches with rb feature. are we testing now that with all benches? just to be sure check == run all.

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (dzmitry-lahoda) [reply .github/workflows/build-test-deploy.yml:966]: oh, these are different.

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -94,34 +65,6 @@ jobs:
         run: |
           git config --global --add safe.directory /__w/composable/composable
 
-      - name: Hadolint
```
[the remark this answers is not in the document]
Comment (andor0) [reply .github/workflows/check.yml:97]: it is disabled now (continue-on-error: true), but it utilizes CPU.  I'm going to move them to nix in the future.

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

File: .github/workflows/check.yml (hunk 2 above)
Comment (andor0) [reply .github/workflows/check.yml:69]: I renamed the existing check from `check-benchmarks` to `test-running-of-pallet-benchmarks`.

**Comment by andor0** [comment]
@mergify refresh

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**